src/mopidy_spotify/web.py

```diff
@@ -64,6 +65,8 @@ def __init__(  # noqa: PLR0913
 
         self._headers = {"Content-Type": "application/json"}
         self._session = utils.get_requests_session(proxy_config or {})
+        self._cache_mutex = Lock()
```

Comment:
What do these protect? It probably doesn't hurt to have a comment by each variable/method that require the mutex to be held saying `Can only be accessed/called while _cache_mutex is held` etc

Reply:
Yes... I think this comment is fair and exposes why this is a bit hacky... Would it helped if I claimed I was going to redo all the caching layers anyway?

Reply:
Yeah, the fact that this base class has a cache mutex which protects something in the child class is a bit off. That is based on my reading of this being that `SpotifyOAuthClient._cache` is what `OAuthClient._cache_mutex` protects, so this does feel like it needs some refactoring or cleanup of sorts.

Reply:
While the refresh case I assume is more a "single request in flight" type scenario which effectively protects `_expires` and `_headers`?

Reply:
Anyway, feel free to postpone addressing this. But perhaps add a todo or something so we reduce the risk of forgetting this?